cluster-provision/gocli/cmd/provision_manager.go

```diff
@@ -0,0 +1,519 @@
+	fmt.Println(string(j))
+
+	return nil
+}
+
+func parseArguments(cmd *cobra.Command) (parameters, error) {
+	params := parameters{}
+	var err error
+
+	params.debug, err = cmd.Flags().GetBool("debug")
+	if err != nil {
+		return parameters{}, err
+	}
+
+	params.rulesFile, err = cmd.Flags().GetString("rules")
+	if err != nil {
+		return parameters{}, err
+	}
+
+	params.tag, err = cmd.Flags().GetString("tag")
+	if err != nil {
+		return parameters{}, err
+	}
+
+	return params, nil
+}
+
+func configLogger(debug bool) {
+	logrus.SetOutput(&OutputSplitter{})
+
+	logrus.SetFormatter(&logrus.TextFormatter{DisableTimestamp: true})
+	if debug {
+		logrus.SetLevel(logrus.DebugLevel)
+		logrus.SetFormatter(&logrus.TextFormatter{DisableLevelTruncation: true, ForceColors: true, DisableTimestamp: true})
+	}
+}
+
+func (splitter *OutputSplitter) Write(p []byte) (n int, err error) {
+	if bytes.Contains(p, []byte("level=error")) || bytes.Contains(p, []byte("level=fatal")) {
+		return os.Stderr.Write(p)
+	}
+	return os.Stdout.Write(p)
+}
+
+func getTargets(path string) ([]string, error) {
+	directories, err := fsys.GlobDirectories("cluster-provision/k8s/*")
+	if err != nil {
+		return nil, err
+	}
+
+	targets := []string{}
+	for _, dir := range directories {
+		targets = append(targets, filepath.Base(dir))
+	}
+
+	logrus.Debug("Targets: ", targets)
+	return targets, nil
+}
+
+func processGitNameStatusChanges(rulesDB map[string][]string, targets []string, tag string) (map[string]bool, error) {
+	cmdOutput, err := runCommand("git", []string{"diff", "--name-status", tag})
+	if err != nil {
+		return nil, err
+	}
+
+	return processChanges(rulesDB, targets, cmdOutput)
+}
+
+func processChanges(rulesDB map[string][]string, targets []string, changes string) (map[string]bool, error) {
+	targetToRebuild := make(map[string]bool)
+	for _, target := range targets {
+		targetToRebuild[target] = false
+	}
+
+	printSection("Changed files")
+
+	errorFound := false
+	files := strings.Split(changes, "\n")
+	for _, nameStatus := range files {
+		if nameStatus == "" {
+			break
+		}
+
+		tokens := strings.Split(nameStatus, "\t")
+		if len(tokens) < 2 {
+			return nil, fmt.Errorf("wrong input syntax, should be <status>\\t<filename>")
+		}
+
+		status := tokens[0]
+		fileName := tokens[1]
+
+		if strings.HasPrefix(status, FILE_RENAMED) {
+			if len(tokens) != 3 {
+				return nil, fmt.Errorf("wrong input syntax, should be <status>\\t<old_filename>\\t<new_filename>")
+			}
+			fileName = tokens[2]
+		}
+
+		// Skip markdown files
+		if strings.HasSuffix(fileName, ".md") {
+			continue
+		}
+
+		match, err := matcher(rulesDB, fileName, status)
+		if err != nil {
+			errorFound = true
+			logrus.Error(err)
+			continue
+		}
+
+		if !errorFound {
+			logrus.Debug(status + " : " + fileName + " - [" + strings.Join(match, " ") + "]")
+		}
+
+		for _, target := range match {
+			if target != TARGET_NONE {
+				targetToRebuild[target] = true
+			}
+		}
+	}
+
+	if errorFound {
```

Comment:
Could we gather the errors that occur ? and log them after this message? 
It could help with troubleshooting - what do you think?

Reply:
This is exactly why we have it and what we are doing
https://github.com/kubevirt/kubevirtci/pull/982/commits/0a205f06aca723a8c48129d29207bb5a97548585#diff-7ed4bf7ea9633967a33dfedd8c17285c330d117c226a8a8b084225ba2518b59eR258